### PR TITLE
Finish code for K64 board

### PR DESCRIPTION
**Pull request by tonerm43**
Code running on board. Seems to be working as far as I can tell without physical hardware

### CODE REVIEW

File: Project_Settings/Debugger/tetris_Debug_Segger.launch [hunk 1]
```diff
@@ -1,38 +1,77 @@
...
 <stringAttribute key="org.eclipse.cdt.debug.gdbjtag.core.jtagDevice" value="GNU ARM J-Link"/>
 <booleanAttribute key="org.eclipse.cdt.debug.gdbjtag.core.loadImage" value="true"/>
 <booleanAttribute key="org.eclipse.cdt.debug.gdbjtag.core.loadSymbols" value="true"/>
+<stringAttribute key="org.eclipse.cdt.debug.gdbjtag.core.pcRegister" value=""/>
 <intAttribute key="org.eclipse.cdt.debug.gdbjtag.core.portNumber" value="2331"/>
 <booleanAttribute key="org.eclipse.cdt.debug.gdbjtag.core.setPcRegister" value="false"/>
 <booleanAttribute key="org.eclipse.cdt.debug.gdbjtag.core.setStopAt" value="true"/>
 <stringAttribute key="org.eclipse.cdt.debug.gdbjtag.core.stopAt" value="main"/>
+<stringAttribute key="org.eclipse.cdt.debug.gdbjtag.core.symbolsFileName" value=""/>
+<stringAttribute key="org.eclipse.cdt.debug.gdbjtag.core.symbolsOffset" value=""/>
```
Comment (brandon1024) [inline Project_Settings/Debugger/tetris_Debug_Segger.launch:55]: Wrong

File: Sources/input-capture.c [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	// Start conversions for ADC
+	ADC0_SC1A |= 0b0000001;
+	ADC1_SC1A |= 0b0000001;
+
+	if ((GPIOC_PDIR >> 10) & 1) {
+		return INPUT_ROTATE;
+	} else if ((GPIOC_PDIR >> 11) & 1) {
+		return INPUT_DROP;
+	} else {
+		// Determine joystick left/right with 'dead zone'
```
Comment (brandon1024) [inline Sources/input-capture.c:40]: sexi comments

File: Sources/input-capture.c [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+		while (!(ADC0_SC1A >> 7 & 1));
+		if (ADC0_RA >= 0x00 && ADC0_RA < 0x60) {
+			// Joystick Left
+			return INPUT_LEFT;
+		} else if (ADC0_RA <= 0xFF && ADC0_RA > 0xA0) {
+			// Joystick Right
+			return INPUT_RIGHT;
+		}
+
+		// Determine joystick up/down with 'dead zone'
```
Comment (brandon1024) [inline Sources/input-capture.c:51]: Wow, you really tried to make this work as much as possible, that's actually impressive

File: Sources/rng.c [hunk 4]
```diff
@@ -1,10 +1,3 @@
-/*
- * rng.c
- *
- *  Created on: Feb 26, 2020
- *      Author: bricha11
- */
-#include "MK64F12.h"
```
Comment (brandon1024) [inline Sources/rng.c:7]: bc fuck "headers"

File: include/rng.h [hunk 5]
```diff
@@ -1,13 +1,8 @@
-/*
```
Comment (brandon1024) [inline include/rng.h:1]: tbh the fact she wants us to write "headers" pisses me off

File: Sources/vga-driver.c [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+			GPIOD_PSOR |= 1;
+			GPIOD_PCOR |= 1;
+		}
+	}
+
+	// Clear GPIO RAM LOAD
+	GPIOC_PCOR |= (1 << 4);
+}
+
+void set_data_lines(uint8_t data) {
```
Comment (brandon1024) [inline Sources/vga-driver.c]: This and the one below can be static

Comment (brandon1024) [reply Sources/vga-driver.c]: but honestly, i dont care hahahah

File: Sources/rng.c (hunk 4 above)
Comment (tonerm43) [reply Sources/rng.c:7]: Fuck do i need to put that back

File: Sources/vga-driver.c [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+	PORTB_PCR9  |= PORT_PCR_MUX(1); // Data bit 2
+	PORTA_PCR1  |= PORT_PCR_MUX(1); // Data bit 3
+	PORTB_PCR23 |= PORT_PCR_MUX(1); // Data bit 4
+	PORTA_PCR2  |= PORT_PCR_MUX(1); // Data bit 5
+	PORTC_PCR2  |= PORT_PCR_MUX(1); // Data bit 6
+	PORTC_PCR3  |= PORT_PCR_MUX(1); // Data bit 7
+
+	GPIOA_PDDR |= ((1 << 1) | (1 << 2)); // Enable pins 1 and 2 for output
+	GPIOB_PDDR |= ((1 << 9) | (1 << 23)); // Enable pins 9 and 23 for output
+	GPIOC_PDDR |= ((1<<2)|(1<<3)|(1<<16)|(1<<17)); // Enable pins 2, 3, 16, 17 for output
```
Comment (brandon1024) [inline Sources/vga-driver.c:34]: Stonk

File: Sources/input-capture.c (hunk 3 above)
Comment (tonerm43) [reply Sources/input-capture.c:51]: I'm sad that we're never gonna actually get to play it tbh

File: Sources/vga-driver.c (hunk 7 above)
Comment (tonerm43) [reply Sources/vga-driver.c:34]: Was too lazy to write it out as hex

**Review by brandon1024 — APPROVED** [review]
Indeed. Very good, I believe that this changelist is important to the syngenesis of our mission, and the freedom of humanity,

File: Sources/input-capture.c (hunk 3 above)
Comment (brandon1024) [reply Sources/input-capture.c:51]: same

File: Sources/rng.c (hunk 4 above)
Comment (brandon1024) [reply Sources/rng.c:7]: :'(

File: Sources/vga-driver.c (hunk 7 above)
Comment (brandon1024) [reply Sources/vga-driver.c:34]: They'll have to nerf you

**Pull request merged**